### PR TITLE
Description of securityMatcher and multiple filter chains has now more details

### PR DESCRIPTION
**Pull request by abimael-turing**
The updated description now provides comprehensive details about the securityMatcher and the concept of multiple filter chains. This enhanced explanation aims to offer a deeper understanding of how these components work together to strengthen security measures in the application.

Closes gh-14419

### CODE REVIEW

**Comment by abimael-turing** [comment]
@sjohnr, I have submitted a pull request. Please review it when you have a moment. Thank you.

File: docs/modules/ROOT/pages/servlet/configuration/java.adoc [hunk 1]
```diff
@@ -250,6 +250,65 @@ public class MultiHttpSecurityConfig {
...
+        return http.build();
+    }
+}
+----
+<1> Begin by configuring authentication settings.
+<2> Define a SecurityFilterChain instance with @Order(1) to handle requests with URLs starting with "/api/". This chain will have the highest priority.
+<3> Specify that the http.securityMatcher applies only to "/api/**" URLs.
+<4> Next, create another SecurityFilterChain instance with @Order(2) to handle requests with URLs starting with "/images/". This chain will be considered second.
+<5> Indicate that the http.securityMatcher applies only to "/images/**" URLs.
+<6> Lastly, create an additional SecurityFilterChain instance without an @Order annotation. This configuration will handle requests not covered by "/api/" or "/images/" URLs and will be processed last (no @Order defaults to last).
```
Comment (sjohnr) [inline docs/modules/ROOT/pages/servlet/configuration/java.adoc:309]: This new example does not seem to provide any new information compared to the previous (existing) example. It simply adds one new pattern with the same authorization rule (`hasRole("ADMIN")`). The new example also doesn't seem to make sense in a real world situation either, as I wouldn't expect an `ADMIN` user to provide credentials via basic authentication in order to access images.

In order to truly improve the documentation, I feel that we need to provide a distinct example of an alternate use case. The new text content we add should correspond directly to the new example.

Would you be able to update this PR to provide a different example and corresponding text content?

File: docs/modules/ROOT/pages/servlet/configuration/java.adoc [hunk 2]
```diff
@@ -203,7 +203,7 @@ Note that this configuration is parallels the XML Namespace configuration:
 
 We can configure multiple `HttpSecurity` instances just as we can have multiple `<http>` blocks in XML.
 The key is to register multiple `SecurityFilterChain` ``@Bean``s.
-The following example has a different configuration for URLs that start with `/api/`.
+The following example has a different configuration for URLs that start with `/api/`.
```
Comment (sjohnr) [inline docs/modules/ROOT/pages/servlet/configuration/java.adoc:206]: This change doesn't seem to be needed. Can you please revert this change?

**Review by sjohnr — CHANGES_REQUESTED** [review]
Thanks for the PR @abimael-turing! However I don't know that this is the change I had in mind with gh-14419. Please see feedback below and let me know if you have any questions.

File: docs/modules/ROOT/pages/servlet/configuration/java.adoc (hunk 2 above)
Comment (abimael-turing) [reply docs/modules/ROOT/pages/servlet/configuration/java.adoc:206]: @sjohnr Thank you, I am working on it!

Comment (abimael-turing) [reply docs/modules/ROOT/pages/servlet/configuration/java.adoc]: @sjohnr, I have implemented the changes as requested. Could you please review this merge request again?

File: docs/modules/ROOT/pages/servlet/configuration/java.adoc (hunk 1 above)
Comment (abimael-turing) [reply docs/modules/ROOT/pages/servlet/configuration/java.adoc]: @sjohnr, I have implemented the changes as requested. Could you please review this merge request again?

**Comment by sjohnr** [comment]
Thanks @abimael-turing and congrats on your first contribution!

I have merged this as 046a1fc811aea379222751c2eddd92a0b12d4e11. However, as-is the change was still very similar to the previous example but was more complex. To address this, I added a few new sections to specifically explain concepts I had in mind with gh-14419, and reworked your example to be a more real-world capstone to the new sections. All of this was added via 
2482e8e446ce8d7dfe41004d981d6fb0e960a648.
